TypeScript 3 update exposed latent issues preventing usage of @loopback/repository-json-schema

## Description / Steps to reproduce / Feature proposal
Try to build a project that uses `@loopback/repository-json-schema` using today's (August 24) release

## Current Behavior
The build fails thus:
```
> lb-tsc es2018

../../common/temp/node_modules/@loopback/repository-json-schema/dist8/src/keys.d.ts:6:75 - error TS2307: Cannot find module '../../context/node_modules/@loopback/metadata/src/types'.

6 export declare const JSON_SCHEMA_KEY: MetadataAccessor<JSONSchema, import("../../context/node_modules/@loopback/metadata/src/types").DecoratorType>;
                                                                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

## Expected Behavior

It should not fail to compile.  It did not fail as of yesterday.

This looks related to https://github.com/strongloop/loopback-next/issues/1405 -- I'm guessing it was introduced by the update to building using TypeScript 3.

## More Notes

The offending release file in today's build:
```
import { MetadataAccessor } from '@loopback/context';
import { JSONSchema6 as JSONSchema } from 'json-schema';
/**
 * Metadata key used to set or retrieve repository JSON Schema
 */
export declare const JSON_SCHEMA_KEY: MetadataAccessor<JSONSchema, import("../../context/node_modules/@loopback/metadata/src/types").DecoratorType>;
```

The same file in the prior release:
```
import { MetadataAccessor } from '@loopback/context';
import { JSONSchema6 as JSONSchema } from 'json-schema';
/**
 * Metadata key used to set or retrieve repository JSON Schema
 */
export declare const JSON_SCHEMA_KEY: MetadataAccessor<JSONSchema, ClassDecorator | PropertyDecorator | MethodDecorator | ParameterDecorator>;
```

I would also note that, AFAICT, importing `MetadataAccessor` from `@loopback/context` is not correct, it should be imported from `@loopback/metadata`.

Experimenting with current `master` (01d3a377), I can reproduce the "bad" build output building it locally.  Changing `repository-json-schema` (`package.json` and `src/types.ts`) to fix that import issue produces a more plausible looking `types.d.ts`:
```
import { MetadataAccessor } from '@loopback/metadata';
import { JSONSchema6 as JSONSchema } from 'json-schema';
/**
 * Metadata key used to set or retrieve repository JSON Schema
 */
export declare const JSON_SCHEMA_KEY: MetadataAccessor<JSONSchema, import("@loopback/metadata/src/types").DecoratorType>;
```

I haven't gone through running tests ~or auditing the code for other instances of this issue~.
A quick grep of the `dist*` folders in a built tree suggests that `packages/authentication/src/keys.ts` and `packages/boot/test/fixtures/application.ts` need this same fix:
```
loopback-next$ find -type d -name dist\* | xargs fgrep -r 'import("../'
./packages/authentication/dist8/src/keys.d.ts:export declare const AUTHENTICATION_METADATA_KEY: MetadataAccessor<AuthenticationMetadata, import("../../context/node_modules/@loopback/metadata/src/types").DecoratorType>;
./packages/authentication/dist10/src/keys.d.ts:export declare const AUTHENTICATION_METADATA_KEY: MetadataAccessor<AuthenticationMetadata, import("../../context/node_modules/@loopback/metadata/src/types").DecoratorType>;
./packages/boot/dist8/test/fixtures/application.d.ts:        bootOptions?: import("../../src/interfaces").BootOptions | undefined;
./packages/boot/dist10/test/fixtures/application.d.ts:        bootOptions?: import("../../src/interfaces").BootOptions | undefined;
```

--- a/packages/authentication/package.json
+++ b/packages/authentication/package.json
@@ -24,6 +24,7 @@
   "license": "MIT",
   "dependencies": {
     "@loopback/context": "^0.12.5",
+    "@loopback/metadata": "^0.9.5",
     "@loopback/core": "^0.11.5",
     "@loopback/dist-util": "^0.3.6",
     "@loopback/openapi-v3": "^0.12.5",

--- a/packages/authentication/src/keys.ts
+++ b/packages/authentication/src/keys.ts
@@ -6,7 +6,8 @@
 import {Strategy} from 'passport';
 import {AuthenticateFn, UserProfile} from './types';
 import {AuthenticationMetadata} from './decorators/authenticate.decorator';
-import {BindingKey, MetadataAccessor} from '@loopback/context';
+import {BindingKey} from '@loopback/context';
+import {MetadataAccessor} from '@loopback/metadata';
 
 /**
  * Binding keys used by this component.

--- a/packages/repository-json-schema/package.json
+++ b/packages/repository-json-schema/package.json
@@ -28,6 +28,7 @@
   },
   "dependencies": {
     "@loopback/context": "^0.12.5",
+    "@loopback/metadata": "^0.9.5",
     "@loopback/dist-util": "^0.3.6",
     "@loopback/repository": "^0.15.1",
     "@types/json-schema": "^6.0.1"

--- a/packages/repository-json-schema/src/keys.ts
+++ b/packages/repository-json-schema/src/keys.ts
@@ -3,7 +3,7 @@
 // This file is licensed under the MIT License.
 // License text available at https://opensource.org/licenses/MIT
 
-import {MetadataAccessor} from '@loopback/context';
+import {MetadataAccessor} from '@loopback/metadata';
 import {JSONSchema6 as JSONSchema} from 'json-schema';
 
 /**

